### PR TITLE
JAMES-2813 Add serialization of Failed, Completed, and Throwing Tasks

### PR DESCRIPTION
**Pull request by blackheaven**

### CODE REVIEW

**Comment by Arsnael** [comment]
```
[ERROR]   ManageableMailQueueContract.browseShouldNotFailWhenConcurrentClearWhenIterating:361 » NoSuchElement
```
test this please

File: server/task-json/src/test/java/org/apache/james/server/task/json/TaskSerializationTest.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
+package org.apache.james.server.task.json;
```
Comment (Arsnael) [inline server/task-json/src/test/java/org/apache/james/server/task/json/TaskSerializationTest.java]: Apache license missing

**Comment by blackheaven** [comment]
Regrouped in #2539

**Comment by blackheaven** [comment]
> do we really need different DTO for these tasks?

Not necessarily for `Completed` and `Failed`, but I find that clearer.

File: server/task-json/src/test/java/org/apache/james/server/task/json/TaskSerializationTest.java (hunk 1 above)
Comment (chibenwa) [reply server/task-json/src/test/java/org/apache/james/server/task/json/TaskSerializationTest.java]: ?

**Comment by blackheaven** [comment]
```
[ERROR]   CassandraMessageIdToImapUidDAOTest.updateShouldUpdateAnsweredFlag:228 
```

test this please

**Comment by rouazana** [comment]
merged

**Pull request closed without merging**